packages/navigation/src/NavigationMenuDropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import type { NavProps } from './Nav';
+import type { CollectionBase } from '@react-types/shared';
+
+import { Chip } from '@launchpad-ui/chip';
+import { Dropdown, DropdownButton } from '@launchpad-ui/dropdown';
+import { Check } from '@launchpad-ui/icons';
+import { Menu, MenuItem } from '@launchpad-ui/menu';
+import { useListState } from '@react-stately/list';
+import { useEffect, useRef, useState } from 'react';
+import { NavLink, useLocation } from 'react-router-dom';
+
+import styles from './styles/Navigation.module.css';
+import { titlecase } from './utils';
+
+type NavigationMenuDropdownProps<T extends object> = CollectionBase<T> & {
+  kind: NavProps['kind'];
+};
+
+const NavigationMenuDropdown = <T extends object>(props: NavigationMenuDropdownProps<T>) => {
+  const state = useListState(props);
+  const { pathname } = useLocation();
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const [buttonWidth, setButtonWidth] = useState<string | undefined>(undefined);
```

Comment:
As the button text changes, the width of the button also changes... we're tracking the state to use it as a `minWidth` on each menu item so that the menu takes is at least as wide as the button (but will grow wider if necessary)